bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexServerHandler.java

```diff
@@ -0,0 +1,306 @@
+                        if (tmpMeta.getType().equals("episode")) {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getGrandparentThumb()));
+                            tmpMeta.setTitle(tmpMeta.getGrandparentTitle() + " : " + tmpMeta.getTitle());
+                        } else if (tmpMeta.getType().equals("track")) {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getThumb()));
+                            tmpMeta.setTitle(tmpMeta.getGrandparentTitle() + " - " + tmpMeta.getParentTitle() + " - "
+                                    + tmpMeta.getTitle());
+                        } else {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getThumb()));
+                        }
+                        playerHandlers.get(tmpMeta.getPlayer().getMachineIdentifier()).refreshSessionData(tmpMeta);
+                        playerActiveCount++;
+                    }
+                }
+            }
+        }
+        updateState(new ChannelUID(getThing().getUID(), PlexBindingConstants.CHANNEL_SERVER_COUNT),
+                new StringType(String.valueOf(playerCount)));
+        updateState(new ChannelUID(getThing().getUID(), PlexBindingConstants.CHANNEL_SERVER_COUNTACTIVE),
+                new StringType(String.valueOf(playerActiveCount)));
+    }
+
+    /**
+     * Refresh all the configured players
+     */
+    private void refreshAllPlayers() {
+        Iterator<PlexPlayerHandler> valueIterator = playerHandlers.values().iterator();
+        while (valueIterator.hasNext()) {
+            valueIterator.next().updateChannels();
+        }
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            int pollingInterval = ((BigDecimal) getConfig().get(CONFIG_REFRESH_RATE)).intValue();
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1, pollingInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * The refresh job, pulls the session data and then calls refreshAllPlayers which will have them send
+     * out their current status.
+     */
+    private Runnable pollingRunnable = () -> {
+        try {
+            MediaContainer plexSessionData = plexAPIConnector.getSessionData();
+            if (plexSessionData != null) {
+                refreshStates(plexSessionData);
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Plex is not returning valid session data");
+            }
+            refreshAllPlayers();
+
+        } catch (Exception e) {
+            logger.warn("An exception occurred while polling the PLEX Server: '{}'", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
The exception's message could be used as the thing status detail message. Then, the log message could be removed and the error is displayed in the UI.